examples/rlhf/models/reward.py

```diff
@@ -0,0 +1,100 @@
+from pathlib import Path
+
+import torch
+import torch.nn as nn
+from tensordict.nn import TensorDictModule
+from transformers import GPT2LMHeadModel, GPT2TokenizerFast
+
+
+class RewardModel(nn.Module):
+    def __init__(self, model_path):
+        super().__init__()
+        model = GPT2LMHeadModel.from_pretrained(model_path, return_dict=False)
+
+        self.config = model.config
+        self.transformer = model.transformer
+
+        # replace last layer with the reward layer
+        self.lm_head = nn.Linear(self.config.n_embd, 1, bias=False)
+        self.PAD_ID = GPT2TokenizerFast.from_pretrained("gpt2").eos_token_id
+
+    def forward(self, input_ids=None, attention_mask=None):
+        outputs = self.transformer(input_ids=input_ids, attention_mask=attention_mask)
+        hidden_states = outputs[0]
+        rewards = self.lm_head(hidden_states).squeeze(-1)
+        end_scores = []
+        bs = input_ids.shape[0]
+
+        for i in range(bs):
+            pad_inds = (input_ids[i] == self.PAD_ID).nonzero()
+            first_pad_ind = (
+                pad_inds[0].item() if len(pad_inds) > 0 else input_ids.shape[1]
+            )
+            end_scores.append(rewards[i, first_pad_ind - 1])
```

Comment:
can't this be batched?
Do you have a minimal example of what this loop does? I'm keen on looking at making it more efficient